src/main/java/spoon/support/compiler/VirtualFolder.java

```diff
@@ -87,9 +87,10 @@ public SpoonFolder getParent() {
 	@Override
 	public List<SpoonFolder> getSubFolders() {
 		List<SpoonFolder> result = new ArrayList<>();
+		Set<SpoonFolder> distinctSet = new HashSet<>();
 		for (SpoonFile f : getAllFiles()) {
 			SpoonFolder folder = f.getParent();
-			if (!result.contains(folder)) {

```

Comment:
The VirtualFolder really miss hashCode, so it would not work correctly with the HashSet. Do you agree to add hashCode method to it?